vllm/compilation/config.py

```diff
@@ -6,12 +6,38 @@
 
 import vllm.envs as envs
 from vllm.logger import init_logger
+from vllm.utils import print_warning_once
 
 from .compile_context import get_compile_context
 
 logger = init_logger(__name__)
 
 
+class PassConfig(BaseModel):
+    """
+    Configuration for custom Inductor passes.
+    This is separate from general CompilationConfig so that inductor passes
+    don't all have access to full configuration - that would create a cycle
+    as the PassManager is set as a property of config.
+    - dump_graph_stages: list of stages for which we want to dump the graph.
+        Each pass defines its own stages (before, after, maybe in-between).
+    - dump_graph_dir: directory to dump the graphs. Default is .
+    - enable_fusion: whether to enable the custom fusion pass.
+    - enable_reshape: whether to enable the custom reshape elimination pass.
+        TODO better pass enabling system.
+    """
+    dump_graph_stages: List[str] = Field(default_factory=list)
+    dump_graph_dir: Path = Field(default=Path("."))
+    enable_fusion: bool = True
+    enable_reshape: bool = True
+
+    def model_post_init(self, __context: Any) -> None:
+        if not self.enable_reshape and self.enable_fusion:
+            print_warning_once(
+                "Fusion enabled but reshape elimination disabled."
+                "RMSNorm + quant (fp8) fusion might not work")
```

Comment:
Is there some reason why you would enable fusion without enabling the reshape pass?  Or just enabling the reshape pass alone?  i.e. why not one flag for both passes (or just call reshape at the end of fusion)

Reply:
Only fp8 fusion needs the reshape pass, int8 doesn't. I was specifically asked to allow disabling the reshape pass but don't feel strongly either way - but if we don't have them decoupled, we should disable fusion if reshape is disabled.

Reply:
I don't have strong feelings either but seems like the two passes work in conjunction and removing redundant reshapes would always be beneficial.

Reply:
@ProExpertProg @bnellnm Because of the compilation time. I found that it increased the compilation time while reshaping 0 ops for bf16 llama.

Reply:
The reshape pass looks dead simple.  Just a linear loop over nodes.  I can't believe it is any more expensive than fix_functionalization or fusion.

Reply:
@bnellnm I agree, so I've been looking into the compilation time. I assume its impact will be negligible, and it's actually the fusion pass and fix_functionalization that are expensive. But I'll report what the compilation times are here

Reply:
Timings are logged (under debug), and there's an option to disable the reshape pass